app/models/name/synonymy.rb

```diff
@@ -180,26 +176,19 @@ def observation_count
   # Returns either self or name,
   # whichever has more observations or was last used.
   def more_popular(name)
-    result = self
-    unless name.deprecated
-      if deprecated
-        result = name
-      elsif observation_count < name.observation_count
-        result = name
```

Comment:
**Question: Did I correctly understand the intent of this method?** (It was not previously tested.):
I think there was a bug:
It did not account for the possibility that `self` had more observations than `name`; instead it immediately fell back on which had the most recent `naming`.